ext/opcache/ZendAccelerator.c

```diff
@@ -505,39 +489,35 @@ zend_string *accel_new_interned_string(zend_string *str)
 	HT_HASH(&ZCSG(interned_strings), nIndex) = HT_IDX_TO_HASH(idx);
 	zend_string_release(str);
 	return p->key;
-#else
-	return str;
-#endif
 }
 
-#ifndef ZTS
 /* Copy PHP interned strings from PHP process memory into the shared memory */
-static void accel_use_shm_interned_strings(void)
+static void accel_copy_permanent_strings(zend_new_interned_string_func_t new_interned_string)
```

Comment:
Just to clarify: Copying to SHM here is just an optimization right? For cross-process purposes it's only important that strings referenced from SHM are interned, not that permanent strings are all in SHM. If something is missed here, it should not cause issues. Is that correct?